webdriver/tests/support/fixtures.py

```diff
@@ -191,7 +194,7 @@ def create_dialog(dialog_type, text=None):
         if text is None:
             text = ""
 
-        assert isinstance(text, basestring), "`text` parameter must be a string"
+        assert isinstance(text, text_type), "`text` parameter must be a string"
```

Comment:
Looks like at least this one is not always unicode in Python 2.

Reply:
This seems the only place. Just updated code. Hope this time it passes wpt.fyi check :).

Reply:
Do we know where non-unicode text was being passed to `create_dialog`? There are many callsites so this may be tricky, but I want to understand why this can be a byte string.

It looks like in terms of what `create_dialog` *does* with `text`, it passes it to `session.execute_async_script` as well as using it compare to `s.alert.text`. In the case of  `session.execute_async_script` it will serialize to JSON I believe (if I'm reading the code right), so that's probably fine (?). But for `alert.text`, isn't that going to always read back unicode in Python3 which would not compare against `text` if someone passed in bytes?

Reply:
I'm not sure this case has much to do with JSON serialization. It's checking the type of arguments that are passed from tests, e.g. in _user_prompts.py_ test file we have
```
@pytest.mark.parametrize("dialog_type, retval", [
    ("alert", None),
    ("confirm", True),
    ("prompt", ""),
])
```
In Python 2, what is given here is _str_ type.

Reply:
My concern was actually passing in *bytes*, not unicode data. For example, if one did:

```
diff --git a/webdriver/tests/add_cookie/user_prompts.py b/webdriver/tests/add_cookie/user_prompts.py
index f58aacd02a..436cbc99e9 100644
--- a/webdriver/tests/add_cookie/user_prompts.py
+++ b/webdriver/tests/add_cookie/user_prompts.py
@@ -23,7 +23,7 @@ def check_user_prompt_closed_without_exception(session, url, create_dialog):
 
         session.url = url("/common/blank.html")
 
-        create_dialog(dialog_type, text=dialog_type)
+        create_dialog(dialog_type, text=b"dialog_byte_text")
 
         response = add_cookie(session, new_cookie)
         assert_success(response)
```

Then what is compared in `wait.until(s.alert.text == text)` (in `create_dialog`) is actually a byte string to a unicode string.

However I overlooked an important point - passing in `text=b"dialog_byte_text"` will only work in Python 2 (where `b"dialog_byte_text"` is a `basestring`), but would throw in Python 3 (where it is **not** a `str`). And in Python 2, this is no worse than what we had before, where we said `basestring` explicitly.

So, LGTM, albeit in Python 2 I think there's a bug here where you can end up comparing across types... but that should be fairly harmless in this case and Python 3 runs will catch it anyway.